ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonGenerator.java

```diff
@@ -48,19 +48,74 @@
 public class IonGenerator
     extends GeneratorBase
 {
+    /**
+     * Enumeration that defines all toggleable features for Ion generators
+     */
+    public enum Feature implements FormatFeature // since 2.12
+    {
+        /**
+         * Whether to use Ion native Type Id construct for indicating type (true);
+         * or "generic" type property (false). Former works better for systems that
+         * are Ion-centric; latter may be better choice for interoperability, when
+         * converting between formats or accepting other formats.
+         *
```

Comment:
May want to mention that this affects writing side (whether to write native type id  or not).

But one thing that should be added is mention of default setting (something like "Enabled by default for backwards-compatibility"?).

Reply:
Okay, done!